partiql-types/src/main/kotlin/org/partiql/errors/TypeCheckException.kt

```diff
@@ -4,3 +4,8 @@ package org.partiql.errors
  * A [TypeCheckException] represents an invalid operation due to argument types.
  */
 public class TypeCheckException : RuntimeException()
+
+/**
+ * A [DataException] represents an unrecoverable query runtime exception.
+ */
+public class DataException(public override val message: String) : RuntimeException()
```

Comment:
Nit. File Name. Also, why is this unrecoverable, i.e., can we catch the exception and return `MISSING` in permissive mode?